database/sql/sql.go

```diff
@@ -190,6 +195,163 @@ func (s *sqlDatabase) cascadeMigration() error {
 	return nil
 }
 
+func (s *sqlDatabase) ensureGithubEndpoint() error {
+	// Create the default Github endpoint.
+	createEndpointParams := params.CreateGithubEndpointParams{
+		Name:          "github.com",
+		Description:   "The github.com endpoint",
```

Comment:
```suggestion
		Name:          "github",
		Description:   "The github endpoint",
```

the `.com` might be a bit confusing in a GitHub-Enterprise context - but no hard meanings on this

Reply:
Before I make changes, I need to add some context around the reasoning behind this.

Right now, the option to use either GitHub or GHES is a part of the credentials we configure in the config. The endpoint is made up of a number of URLs for the base URL )used to compute repo/org/enterprise URLs that are sent to the runners when token config is used), API URL and upload URLs. There is no other info about the endpoint itself.

The assumption was made that there is an extremely low chance that more than one GH installation will use the same exact hostname, but different URL paths. So during the migration of credentials from the config to the DB, it made sense to use the hostname as a `name` for the new DB entry to uniquely identify the endpoint. The name in the DB is also the primary key.

The `github.com` endpoint is the default endpoint that will always exist and which cannot be removed.

I used `github.com` here instead of `github` to be consistent with how a potential GHES endpoint would look like post-migration. In most cases this will look something like `ghes.example.com` or `git.example.com`. But I am too close to the code at this point and may have failed to consider potential confusion. If given the above, you feel that seeing `github.com` instead of `github` as a unique identifier for this endpoint is confusing, I will gladly change it :smile:

Reply:
I should clarify. When using GHES, a new endpoint will need to be added (if starting from scratch) using:

```bash
garm-cli github endpoint create \
    --base-url https://ghes.example.com \
    --api-base-url https://api.ghes.example.com \
    --upload-url https://upload.ghes.example.com \
    --name example \
    --description "Just an example ghes endpoint" \
    --ca-cert-path $HOME/ca-cert.pem
```

Then you will need to add credentials associated with this new endpoint:

```bash
# for App credential type
garm-cli github credentials add \
    --name deleteme \
    --description "just a test" \
    --endpoint example \
    --auth-type app \
    --app-id 1 \
    --app-installation-id 99 \
    --private-key-path /home/ubuntu/sample.txt
```

```bash
garm-cli github credentials add \
    --name deleteme \
    --description "just a test" \
    --auth-type pat \
    --pat-oauth-token gh_someToken \
    --endpoint example
```

Reply:
ignore my comment - just stumpled over `github.com` as string and was confused. after a first test in our setup, i've seen that this `github.com` string appears in the default endpoint configuration:

The github Endpoint configuration is stored in a dedicated table: (`select * from github_endpoints;`)

| name|created_at|updated_at|deleted_at|description|api_base_url|upload_base_url|base_url|ca_cert_bundle
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
github.com|2024-05-07 11:34:12.171717827+00:00|2024-05-07 11:34:12.171717827+00:00||The github.com endpoint|https://api.github.com/|https://uploads.github.com/|https://github.com|
git.company.com |2024-05-07 11:34:12.172863756+00:00|2024-05-07 11:34:12.172863756+00:00||Endpoint for git.company.com |https://git.company.com |https://git.company.com |https://git.company.com |